File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/PermissionsListener.java
package com.mapbox.maps.plugin.locationcomponent;

import java.util.List;

/**
 * Callback used in PermissionsManager
 */

public interface PermissionsListener {

    void onExplanationNeeded(List<String> permissionsToExplain);

    void onPermissionResult(boolean granted);
}

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/PermissionsListener.kt
package com.mapbox.maps.plugin.locationcomponent

/**
 * Callback used in PermissionsManager
 */
interface PermissionsListener {
  /**
   * Called when an explanation to the user needs to be shown
   */
  fun onExplanationNeeded(permissionsToExplain: List<String>)

  /**
   * Called when permissions have been granted or denied
   */
  fun onPermissionResult(granted: Boolean)
}